module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,66 @@
+#include "compression.hpp"
+#include <forward_list>
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height>& bitmap)
+{
+    std::vector<std::pair<uint8_t, uint8_t>> compression { };
+
+    for(auto it : bitmap)
+    {
+        compression.emplace_back(std::make_pair(it.front(), 0));
+        for(auto it2 : it)
+        {
+            std::pair<uint8_t, uint8_t>& pair = compression.back();
+            if(it2 == pair.first)
+            {
+                
+                pair.second++;
+            }
+            else
+            {
+                compression.emplace_back(std::make_pair(it2, 1));
+            }
+        }
+    }
+
+    return compression;
+}
```

Comment:
You can do this with using emplace_back only once :)

Reply:
ye but it probably needs more tweaking right? ;p